pkg/collector/python/helpers.go

```diff
@@ -126,13 +127,17 @@ func (sl *stickyLock) unlock() {
 func cStringArrayToSlice(array **C.char) []string {
 	if array != nil {
 		res := []string{}
+		si := sharedStringInternerPool.Get()
+		defer sharedStringInternerPool.Put(si)
```

Comment:
Note: Not sure if a sync.Pool is the right choice here. I suspect a single interner behind a (RW)Mutex would be a better choice. What's the fastest way to experiment with this? Are there any local benchmarks I could run?